script/color-contrast.ts

```diff
@@ -97,76 +104,80 @@ const testContrast = (minimumContrast: number, colorA: string, colorB: string, b
+ * @param theme
+ * @param results
  */
 const renderConsoleTable = (theme: string, results: contrastTestResult[]): void => {
   // config table
   const contrastTable = new Table({
     title: `Contrast checks for: ${theme}`,
-    charLength: { "❌": 2, "✅": 2 },
+    charLength: {'❌': 2, '✅': 2},
     colorMap: {
-      grey: '\x1b[0;30m', // define customized color
+      grey: '\x1b[0;30m' // define customized color
     },
     columns: [
       {
-        name: "contrastPair",
-        alignment: "left",
-        title: "Color pair",
+        name: 'contrastPair',
+        alignment: 'left',
+        title: 'Color pair'
       },
       {
-        name: "pass",
-        alignment: "center",
-        title: "Pass",
+        name: 'pass',
+        alignment: 'center',
+        title: 'Pass'
       },
       {
-        name: "contrastRatio",
-        alignment: "left",
-        title: "Contrast ratio"
+        name: 'contrastRatio',
+        alignment: 'left',
+        title: 'Contrast ratio'
       },
       {
-        name: "minimumContrastRatio",
-        alignment: "left",
-        title: "Min. ratio"
-      }]
+        name: 'minimumContrastRatio',
+        alignment: 'left',
+        title: 'Min. ratio'
+      }
+    ]
   })
   // add rows and color
   for (const [index, row] of results.entries()) {
-    let color = "white"
+    let color = 'white'
     // turn odd index rows grey (index starts at 0)
     if (index % 2 !== 0) {
-      color = "grey"
+      color = 'grey'
     }
     contrastTable.addRow(row, {
-      color: color
+      color
     })
   }
   // print table to console
+  // eslint-disable-next-line no-console
   console.log(contrastTable.render())
 }
+
 /**
  * run tests, output results to console and store them for json
  */
-const results = Object.entries(contrastRequirements)
-  .map(([theme, colorPairs]: [theme: string, colorPairs: ContrastRequirement[]]) => {
-    // turn deeply nested colors object into one level object like 'fg.default': '#000'
-    // @ts-ignore
-    const flattenColors = flattenObject(colors.colors[theme])
-    // run tests on all color pairs
-    const results = runContrastTest(colorPairs, flattenColors)
-    // print results to console
-    renderConsoleTable(theme, results)
-    // return results for json file creation
-    return {
-      theme,
-      results
-    }
-  })
+const results = Object.entries(contrastRequirements).map(([theme, colorPairs]: [string, ContrastRequirement[]]) => {
+  // turn deeply nested colors object into one level object like 'fg.default': '#000'
 
-// write json file for workflow
-fs.writeFile('dist/color-contrast-check.json', JSON.stringify(results), (err) => {
-  if (err) throw err;
-  console.log('The file has been saved!');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const flattenColors = flattenObject((colors.colors as any)[theme])
```

Comment:
To patch the typescript error, I've cast this any for now. Hopefully we won't have this with new tokens that have their own type definitions. This is an example of where we'd want stricter typing of our JSON @lukasoppermann.

Reply:
Yes, but we would have to use `string` or `colorHex` with the new tokens, right? As we can't add types to the json directly. So we still need to cast, just not `any`.

Reply:
yes, any is here to help remove the `@ts-ignore` that was previously here. Ideally this JSON source would be strongly typed like an API contract, which we can cast immediately after importing.

Reply:
Is this possible to make the JSON strongly type or do you mean by enforcing it through `eslint`?

Reply:
I imagine it to work like an API response, where we can provide a corresponding typed interface that is co-located to the JSON output, which either tsc can pick up at the same time or we import manually as cast against the import.

Reply:
Did you do this before? I am not really sure how this would work. Would we just hand-type a file and assume in `colors/*.json` are only color tokens. Or we could allow all types of tokens.

I am happy to work on this but it would be great if you could provide an example here so I am moving in the right direction.

Reply:
Yeah it's essentially the same thing style-dictionary already does for its own built-in typed interfaces. Instead of using it for CommonJS modules we use it for JSON. The Style Dictionary version is actually just stringified JSON, so it'll work out of the box. I can show you an example of this later separately.